lib/reek/context/code_context.rb

```diff
@@ -16,13 +15,16 @@ module Context
     #
     # :reek:TooManyMethods: { max_methods: 19 }
     # :reek:TooManyInstanceVariables: { max_instance_variables: 8 }
+    # :reek:Attribute
     class CodeContext
       extend Forwardable
       delegate each_node: :exp
       delegate %i(name type) => :exp
-      delegate %i(visibility visibility= non_public_visibility?) => :visibility_tracker
 
-      attr_reader :children, :context, :exp, :statement_counter, :visibility_tracker
+      attr_accessor :visibility
```

Comment:
Ok, from what I gather reading is that you are using a `visibility` attribute on `CodeContext` that then the `VisibilityTracker` in all inherited contexts uses, right?
Personally I find this confusing. Can't we just put all-the-things in `VisibilityTracker`?